### ISSUE TITLE
Handling auto-generated protobuf files

### ISSUE THREAD

**Issue by dbwiddis** [opensearch-project/opensearch-sdk-py#23]
While reviewing our [code coverage](https://app.codecov.io/gh/opensearch-project/opensearch-sdk-py/tree/main/src%2Fopensearch_sdk_py) I noticed the majority of our uncovered lines (57 of 102) are in the `protobuf` subdirectory.

Initial question: should we exclude these files from coverage (I think so).

Follow-up question: Where else should (or shouldn't) we exclude the auto-generated files? 
 - Should they be excluded from being committed (.gitignore) so we don't need to worry about it?
   - That puts a requirement on the user to generate them, which we shouldn't require unless it's built into how they run it (e.g., poetry) so I'm happy with them being in the repo.  Plus we don't change them often.
 - I excluded from formatting because they said "do not edit" but it doesn't hurt to format them.
 - It looks like they aren't excluded from license headers.  I briefly looked into this and generated files generally are considered an "object" with the same license as the source (much like compiled code) so it doesn't hurt to add the license here.
    - But since we're "modifying" the file to add the license header, perhaps we should just go ahead and modify them with formatting scripts.

TLDR:
 - I think we should exclude proto files from code coverage
 - I think we should remove the exclusion from formatting tools

**Comment by dblock** [comment]
I think the question is whether these files are part of source code, or part of generated files at build time. Your tl;dr says it's the latter, which means we should _build_ them as part of `poetry install` or something like that.

**Comment by dbwiddis** [comment]
An added complication of build time, however, is that `protoc` generates proto3-incompatible import statments, see https://github.com/protocolbuffers/protobuf/issues/1491.  I eventually figured out that `2to3` is one compatible workaround.

Given how little (if ever) these files are going to change, I'm fine with pre-building them. It's more of a question of how much we should touch them with additional automation.

**Comment by dblock** [comment]
IMO, since these are generated files we should make sure they have a *this is generated* header and otherwise not touch them.

**Comment by dblock** [comment]
In https://github.com/opensearch-project/opensearch-sdk-py/pull/54#discussion_r1335357156 we suggest removing the lines of generating proto files from CI and documenting how to update them.
